### PR TITLE
Created tutorial for tf.data and tf.estimator

### PR DESCRIPTION
**Pull request by peastman**

### CODE REVIEW

**Comment by rbharath** [comment]
@peastman This looks amazing! Really explains the inter-op well.

@ankitvgupta Would be great to hear your thoughts on the tutorial. Would this have been useful when your team was looking into deploying DeepChem models?

**Comment by ankitvgupta** [comment]
Looks great! Nicely done @peastman 

I did want to give one suggestion though. I believe it is not true that the `tf.metrics.auc` only allows for a single output, as that writeup suggested when it says "It provides an AUC metric, but no easy way to average this metric over tasks". In fact, the labels, predictions, and weights can all be of arbitrary (but all the same) shape, and it will calculate the AUC averaged across all of them. I believe it is calculated as though you called `tf.layers.flatten()` on each of those tensors, and we use that in production.

In any case, looks great and yes would have been useful (and likely will continue to be for us).

**Comment by peastman** [comment]
Thanks!  I'll make that change, since it will certainly simplify the example.

Is that behavior documented somewhere?  I don't see any mention of it at https://www.tensorflow.org/api_docs/python/tf/metrics/auc.

**Comment by ankitvgupta** [comment]
Sure thing! Yeah, in that link you posted, you'll note that in the parameter description, it says

```
labels: A Tensor whose shape matches predictions. Will be cast to bool.
predictions: A floating point Tensor of arbitrary shape and whose values are in the range [0, 1].
```
What it doesn't say is exactly what it does. To find, you have to dig through the tensorflow code a bit, and I think this is the line where the flattening happens when computing the confusion matrix: https://github.com/tensorflow/tensorflow/blob/r1.6/tensorflow/python/ops/metrics_impl.py#L484

**Comment by peastman** [comment]
That looks like it's doing something very different.  It's just taking your predictions and labels and flattening them into a single list.  Instead of computing AUC separately for each task, it pretends everything is just one task and looks at the effect of using a single cutoff on everything all at once.  If all the tasks just happen to have similar output distributions, that might give a result that's not too far from the correct one.  But if their distributions differ significantly, you'll get something totally wrong.

**Comment by rbharath** [comment]
@peastman Is this PR ready to merge in?

**Comment by ankitvgupta** [comment]
Totally agreed that it’s not averaging the AUCs, but instead treating it as one task. I still think, though, that it may be useful to users to know about that option.

**Comment by peastman** [comment]
In that case, let's leave the tutorial as it is.  People are likely to copy the code in the tutorial, so I want to make sure they'll be doing the calculation correctly.  I just checked in a few minor changes.  Once the tests pass it should be ready to merge.

**Comment by rbharath** [comment]
LGTM
